src/utils/promiseAllSettled.js

```diff
@@ -0,0 +1,18 @@
+/**
+ *
+ * @param { [Promise<T>] } promises
+ * @returns {Promise<[{ status: "fulfilled", value: T} | { status: "rejected", reason: Error}]> }
+ * @template T
+ */
+
+function allSettled(promises) {
```

Comment:
Nice. We should actually bump the requirement on Node to 12 at this point, but that's outside the scope of this PR.